runtime/drivers/blob/blobdownloader.go

```diff
@@ -138,6 +155,10 @@ func NewIterator(ctx context.Context, bucket *blob.Bucket, opts Options, l *zap.
 
 // Close frees the resources
 func (it *blobIterator) Close() error {
+	it.cancel()
+	if it.grp != nil {
+		_ = it.grp.Wait() // wait for background calls to complete
+	}
 	// remove temp dir since recursive paths created have to be removed as well
 	err := os.RemoveAll(it.tempDir)
```

Comment:
Since `Close` may be called on line 133, would suggest:
```suggestion
	// remove temp dir since recursive paths created have to be removed as well
	var err error
	if it.tempDir != "" {
		err = os.RemoveAll(it.tempDir)
	}
```